File: CS-Examples/21_Security/AddDigitalSignature.cs
using Spire.Xls;
using Spire.Xls.Core.MergeSpreadsheet.Interfaces;
using System;
using System.Security.Cryptography.X509Certificates;
using System.Windows.Forms;

namespace AddDigitalSignature
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a workbook
            Workbook workbook = new Workbook();

            // Load file from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\DigitalSignature.xlsx");

            // Add a digital certificate for signing the workbook
            String inputFile_pfx = @"..\..\..\..\..\..\Data\gary.pfx";            
            X509Certificate2 cert = new X509Certificate2(inputFile_pfx, "e-iceblue");

            // Specify the date and time for the digital signature
            DateTime certtime = new DateTime(2020, 7, 1, 7, 10, 36);

            // Add a digital signature to the workbook using the provided certificate, signer name ("e-iceblue"), and signature timestamp
            IDigitalSignatures dsc = workbook.AddDigitalSignature(cert, "e-iceblue", certtime);

            // Specify the output filename for the signed workbook
            String result = "AddDigitalSignature.xlsx";

            // Save the workbook with the added digital signature to the specified file in Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: VB-Examples/21_Security/AddDigitalSignature.vb
Imports Spire.Xls
Imports Spire.Xls.Core.MergeSpreadsheet.Interfaces
Imports System.Security.Cryptography.X509Certificates

Namespace AddDigitalSignature
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the Excel file from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\DigitalSignature.xlsx")

            ' Specify the path of the input PFX file
            Dim inputFile_pfx As String = "..\..\..\..\..\..\Data\gary.pfx"

            ' Create a new X509Certificate2 object using the PFX file and password
            Dim cert As New X509Certificate2(inputFile_pfx, "e-iceblue")

            ' Specify the date and time for the digital signature
            Dim certtime As New Date(2020, 7, 1, 7, 10, 36)

            ' Add a digital signature to the Workbook using the specified certificate, reason, and time
            Dim dsc As IDigitalSignatures = workbook.AddDigitalSignature(cert, "e-iceblue", certtime)

            ' Specify the name of the resulting Excel file after adding the digital signature
            Dim result As String = "AddDigitalSignature.xlsx"

            ' Save the Workbook to the specified path in Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010)

            ' Release the resources used by the workbook
            workbook.Dispose()
            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
